### PR TITLE
Grib processed data

### PR DESCRIPTION
**Pull request by marqh**
handle type of processed data for GRIB export

### CODE REVIEW

File: lib/iris/fileformats/grib/_save_rules.py [hunk 1]
```diff
@@ -164,9 +164,16 @@ def identification(cube, grib):
     # operational product, operational test, research product, etc
     # (missing for now)
     gribapi.grib_set_long(grib, "productionStatusOfProcessedData", 255)
+
+    # Code table 1.4
     # analysis, forecast, processed satellite, processed radar,
-    # (analysis and forecast products for now)
-    gribapi.grib_set_long(grib, "typeOfProcessedData", 2)
+    if cube.coords('realization') and len(cube.coords('realization')) == 1:
```
Comment (bjlittle) [inline lib/iris/fileformats/grib/_save_rules.py]: @marqh I've never seen a cube with more than one `realization` coordinate. What use case are you targeting here, where a cube could contain more than one `realization` coordinate?

File: lib/iris/fileformats/grib/_save_rules.py [hunk 2]
```diff
@@ -164,9 +164,16 @@ def identification(cube, grib):
...
+        if cube.coord('realization').points != (0,):
+            gribapi.grib_set_long(grib, "typeOfProcessedData", 4)
+        else:
+            gribapi.grib_set_long(grib, "typeOfProcessedData", 3)
```
Comment (bjlittle) [inline lib/iris/fileformats/grib/_save_rules.py]: @marqh Fancy adding a minor comment to explain the meaning of the `typeOfProcessedData` values ... or simply reference the GRIB code table, now that would be handy? I'm guessing it's **Code table 4.3** ?

File: lib/iris/fileformats/grib/_save_rules.py (hunk 1 above)
Comment (marqh) [inline lib/iris/fileformats/grib/_save_rules.py]: it's not checking for many realization coordinates, it's trying to look for one.
```
cube.coord('realization')
```
throws an exception if there are none, which is a bit messy in the code flow

File: lib/iris/fileformats/grib/_save_rules.py (hunk 2 above)
Comment (marqh) [inline lib/iris/fileformats/grib/_save_rules.py]: it is referenced, on line 168, it is taken from code table 1.4

Comment (bjlittle) [inline lib/iris/fileformats/grib/_save_rules.py]: Yup, thanks, clearly I can't read! Apologies :smile:

File: lib/iris/fileformats/grib/_save_rules.py (hunk 1 above)
Comment (bjlittle) [inline lib/iris/fileformats/grib/_save_rules.py]:
```python
if cube.coords('realization'):
```
will simply do the trick!

**Comment by bjlittle** [comment]
@marqh Minor tweak and we're there :+1:

**Pull request merged**